### DISCUSSION TITLE
How to run the same function with different memorySize settings?

### DISCUSSION THREAD

**Discussion by weavermedia** [serverless/serverless#11655] [Q&A]
I have a simple JS function that runs an FFMPEG command. In my `serverless.yml` I assign `memorySize: 10240` so it will run as fast as possible for the user.

I now also want to run the same function but with a smaller memory size, say 1000, for users on the free plan who don't get high-speed encoding.

How do I run the same function with a different memory size without duplicating the whole thing?

**Comment by pgrzesik** [comment]
Hey @weavermedia - at the moment AWS doesn't offer support for such dynamic configuration for Lambda functions. The only solution is to deploy a duplicate function with different memory settings. 

Similar problem has been described here (it's ephemeral storage size, not memory, but the general idea is the same): https://medium.com/storyboard-podcasts/storyboard-dev-blog-serverless-compute-with-dynamic-ephemeral-storage-8fd38c0df93d
